### DISCUSSION TITLE
How to broadcast events using zustand?

### DISCUSSION THREAD

**Discussion by hexwit** [pmndrs/zustand#2021] [Q&A]
Preface: 
app use zustand for state managing, business logic and calling API Client. It has no any relation to UI. 

Problem:
In the situation when browser is offline, I receive network error from API Client. 
And I would like to broadcast it to a special component that will render message about offline. 
I don't want to add ui stuff to the store. Also, I don't need to store error in the state. So need just to broadcast it to subscribers. 

The question: is there any valid solution how I can achieve my goal, broadcast a message using zustand? 
May be you could suggest some alternatives to this approach.

**Comment by TimKieu** [comment]
I am also looking dispatch/broadcast mechansm like RTK.
And try to use on i18n without reloading pages. 
We may apply UI techniques (React/Next/Qwik..) not zustand ones.
